x-pack/plugins/security/public/views/management/edit_role/components/privileges/kibana/space_aware_privilege_section/space_aware_privilege_section.tsx

```diff
@@ -0,0 +1,290 @@
+import { PrivilegeDefinition, Role } from '../../../../../../../../common/model';
+import { EffectivePrivilegesFactory } from '../../../../../../../lib/effective_privileges';
+import { RoleValidator } from '../../../../lib/validate_role';
+import { PrivilegeMatrix } from './privilege_matrix';
+import { PrivilegeSpaceForm } from './privilege_space_form';
+import { PrivilegeSpaceTable } from './privilege_space_table';
+
+interface Props {
+  privilegeDefinition: PrivilegeDefinition;
+  role: Role;
+  effectivePrivilegesFactory: EffectivePrivilegesFactory;
+  spaces: Space[];
+  onChange: (role: Role) => void;
+  editable: boolean;
+  validator: RoleValidator;
+  intl: InjectedIntl;
+  uiCapabilities: UICapabilities;
+  features: Feature[];
+}
+
+interface State {
+  role: Role | null;
+  editingIndex: number;
+  showSpacePrivilegeEditor: boolean;
+  showPrivilegeMatrix: boolean;
+}
+
+class SpaceAwarePrivilegeSectionUI extends Component<Props, State> {
+  private globalSpaceEntry: Space = {
+    id: '*',
+    name: this.props.intl.formatMessage({
+      id: 'xpack.security.management.editRole.spaceAwarePrivilegeForm.globalSpacesName',
+      defaultMessage: '* Global (all spaces)',
+    }),
+    color: '#D3DAE6',
+    initials: '*',
+    disabledFeatures: [],
+  };
+
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      showSpacePrivilegeEditor: false,
+      showPrivilegeMatrix: false,
+      role: null,
+      editingIndex: -1,
+    };
+  }
+
+  public render() {
+    const { uiCapabilities, effectivePrivilegesFactory } = this.props;
+
+    if (!uiCapabilities.spaces.manage) {
+      return (
+        <EuiCallOut
+          title={
+            <p>
+              <FormattedMessage
+                id="xpack.security.management.editRole.spaceAwarePrivilegeForm.insufficientPrivilegesDescription"
+                defaultMessage="Insufficient Privileges"
+              />
+            </p>
+          }
+          iconType="alert"
+          color="danger"
+          data-test-subj="userCannotManageSpacesCallout"
+        >
+          <p>
+            <FormattedMessage
+              id="xpack.security.management.editRole.spaceAwarePrivilegeForm.howToViewAllAvailableSpacesDescription"
+              defaultMessage="You are not authorized to view all available spaces."
+            />
+          </p>
+          <p>
+            <FormattedMessage
+              id="xpack.security.management.editRole.spaceAwarePrivilegeForm.ensureAccountHasAllPrivilegesGrantedDescription"
+              defaultMessage="Please ensure your account has all privileges granted by the
+              {kibanaUser} role, and try again."
+              values={{
+                kibanaUser: (
+                  <strong>
+                    <FormattedMessage
+                      id="xpack.security.management.editRole.spaceAwarePrivilegeForm.kibanaUserTitle"
+                      defaultMessage="kibana_user"
```

Comment:
You're probably right, but I'm tempted to leave this alone for now because it's not part of my PR, and was done by the `i18n` team. It shows up as an addition because I moved this entire file.